yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -122,10 +122,14 @@ export class AvmPersistableStateManager {
    * @returns true if the note hash exists at the given leaf index, false otherwise
    */
   public async checkNoteHashExists(storageAddress: Fr, noteHash: Fr, leafIndex: Fr): Promise<boolean> {
-    const gotLeafIndex = await this.hostStorage.commitmentsDb.getCommitmentIndex(noteHash);
-    const exists = gotLeafIndex === leafIndex.toBigInt();
-    this.log.debug(`noteHashes(${storageAddress})@${noteHash} ?? leafIndex: ${leafIndex}, exists: ${exists}.`);
-    this.trace.traceNoteHashCheck(storageAddress, noteHash, leafIndex, exists);
+    const gotLeafValue = await this.hostStorage.commitmentsDb.getCommitmentValue(leafIndex.toBigInt());
```

Comment:
We now get the value given the index rather than the other way around, if the values do not match, we return the value that we found (this way the kernel can verify the merkle path).

We constrain equality in the VM (eventually)